### DISCUSSION TITLE
Importing hf5 file

### DISCUSSION THREAD

**Discussion by AdrienBerger38** [mtex-toolbox/mtex#1084] [Ask Anything]
Hi,

I am sorry to open an issue for such as ridiculous request, but I'm new to both Matlab and MTEX and I'm having a lot of trouble importing a simple hf5 file.

My EBSD data is saved in a .h5 file with shape (height, width, 3) corresponding to euler angles in Bunge convention in every pixel. The material is BCC ferrite. Non-indexed points carry a numpy.nan value (the file was created in python).

Could you please tell me what lines I should write to import the data?
Many thanks in advance.

Best regards,
Adrien Berger

**Comment by kilir** [comment]
Hi,
with recent versions of Matlab you can use `h5info()` to find the groups/fields and `h5read()`. Is it a custom file or something produced by a more commonly used machine?

Cheers,
Rüdiger

**Reply by AdrienBerger38** [reply]
Thanks for your answer.

Here is the file information:


struct with fields:

      Filename: '/home/adrien.berger/dic_ebsd.h5'
          Name: '/'
        Groups: []
      Datasets: [1×1 struct]
     Datatypes: []
         Links: []
    Attributes: []


It is a custom file created by me. I can create it again or modify it if need be.
Best regards,
Adrien

**Comment by AdrienBerger38** [comment]
Update: I managed to load my data as a .txt file.
The issue is that the transformation from .npy to .txt is much slower, but at least it works fine.

I guess the issue can be closed.

Best regards,
Adrien

**Reply by ralfHielscher** [reply]
If you know the structure of your hf5 file it should be simple for you to extract the the orientations, the spatial x,y coordinates and the phase information and construct an MTEX EBSD variable using the syntax
```matlab
prop.xy = xy;
prop.customProperty = myProperty;
CS = crystalSymmetry('mySymmetry')
CSList = {'notIndexed',CS};
rot = rotation.byEuler(phi1,Phi,phi2);
ebsd = EBSD(rot,phaseId,CSList,prop)
```
see also https://mtex-toolbox.github.io/EBSD.EBSD.html
Ralf.
